# STEP 6 — Machine Learning Data Preparation

## Objective

The objective of this step is to prepare the feature-engineered population dataset for machine learning.

### Main Tasks

- Load the feature-engineered dataset
- Select the input features
- Define the target variable
- Split the data into training and testing sets using a time-based split

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv(
    "../data/processed/Global_Population_Feature_Engineered.csv"
)

print("Dataset Shape:", df.shape)
df.head()

Dataset Shape: (13858, 10)


,Country Name,Country Code,Indicator Name,Indicator Code,Year,Population,Previous_Year_Population,Population_Growth_Percentage,Population_Lag_2,Population_Rolling_Avg_3
0,Aruba,ABW,"Population, total",SP.POP.TOTL,1962,56320.0,55578.0,1.335061,54922.0,55606.666667
1,Aruba,ABW,"Population, total",SP.POP.TOTL,1963,57002.0,56320.0,1.210938,55578.0,56300.000000
2,Aruba,ABW,"Population, total",SP.POP.TOTL,1964,57619.0,57002.0,1.082418,56320.0,56980.333333
3,Aruba,ABW,"Population, total",SP.POP.TOTL,1965,58190.0,57619.0,0.990993,57002.0,57603.666667
4,Aruba,ABW,"Population, total",SP.POP.TOTL,1966,58694.0,58190.0,0.866128,57619.0,58167.666667


In [3]:
print("Minimum Year:", df["Year"].min())
print("Maximum Year:", df["Year"].max())
print("Number of Countries:", df["Country Code"].nunique())

Minimum Year: 1962
Maximum Year: 2025
Number of Countries: 217


In [4]:
features = [
    "Year",
    "Previous_Year_Population",
    "Population_Growth_Percentage",
    "Population_Lag_2",
    "Population_Rolling_Avg_3"
]

target = "Population"

X = df[features]
y = df[target]

print("Features:")
print(features)

print("\nTarget:")
print(target)

print("\nX Shape:", X.shape)
print("y Shape:", y.shape)

Features:
['Year', 'Previous_Year_Population', 'Population_Growth_Percentage', 'Population_Lag_2', 'Population_Rolling_Avg_3']

Target:
Population

X Shape: (13858, 5)
y Shape: (13858,)


In [5]:
train_df = df[df["Year"] <= 2020].copy()
test_df = df[df["Year"] > 2020].copy()

print("Training Data:", train_df.shape)
print("Testing Data:", test_df.shape)

print("\nTraining Years:",
      train_df["Year"].min(),
      "-",
      train_df["Year"].max())

print("Testing Years:",
      test_df["Year"].min(),
      "-",
      test_df["Year"].max())

Training Data: (12773, 10)
Testing Data: (1085, 10)

Training Years: 1962 - 2020
Testing Years: 2021 - 2025


In [6]:
X_train = train_df[features]
y_train = train_df[target]

X_test = test_df[features]
y_test = test_df[target]

print("X_train:", X_train.shape)
print("y_train:", y_train.shape)

print("X_test:", X_test.shape)
print("y_test:", y_test.shape)

X_train: (12773, 5)
y_train: (12773,)
X_test: (1085, 5)
y_test: (1085,)


## Data Preparation Summary

The feature-engineered population dataset was divided into training and testing datasets using a time-based split.

### Training Dataset
- Period: 1962–2020
- Records: 12,773
- Features: 5

### Testing Dataset
- Period: 2021–2025
- Records: 1,085
- Features: 5

### Input Features

1. Year
2. Previous Year Population
3. Population Growth Percentage
4. Population Lag 2
5. Population Rolling Average (3 years)

### Target Variable

- Population

A time-based split was used to prevent future information from being used during model training.